### PR TITLE
chore(liveness): adding a11y error label to alert icon

### PR DESCRIPTION
**Pull request by esauerbo**
#### Description of changes
Added `aria-label="Error"` to Timeout Warning `AlertIcon` to indicate that it's an error
Screenshot 2024-01-05 at 12 00 36 PM 1

 

#### Description of how you validated changes
Ran through VoiceOver; 'Error' text is announced 

#### Checklist

- [ ] Have read the [Pull Request Guidelines](https://github.com/aws-amplify/amplify-ui/blob/main/CONTRIBUTING.md)
- [ ] PR description included
- [ ] Relevant documentation is changed or added (and PR referenced)
- [ ] `yarn test` passes and tests are updated/added
- [ ] No side effects or [`sideEffects`](https://github.com/aws-amplify/amplify-ui/blob/main/packages/react/CONTRIBUTING.md#code-standards) field updated

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/react-liveness/src/components/FaceLivenessDetector/shared/FaceLivenessErrorModal.tsx [hunk 1]
```diff
@@ -68,7 +68,7 @@ const renderToastErrorModal = (props: {
   return (
     <>
       <Flex className={LivenessClassNames.ErrorModal}>
-        <AlertIcon ariaHidden variation="error" />
+        <AlertIcon aria-label="Error" role="img" variation="error" />
```
Comment (hbuchel) [inline packages/react-liveness/src/components/FaceLivenessDetector/shared/FaceLivenessErrorModal.tsx:71]: `Error` should be translate-able

File: packages/react/src/primitives/Alert/AlertIcon.tsx [hunk 2]
```diff
@@ -22,26 +23,37 @@ interface AlertIconProps {
...
+      );
       break;
   }
 
   return icon ? (
-    <span className={ComponentClassName.AlertIcon}>{icon}</span>
+    <span
+      className={ComponentClassName.AlertIcon}
+      aria-hidden={ariaHidden}
+      {...rest}
```
Comment (hbuchel) [inline packages/react/src/primitives/Alert/AlertIcon.tsx:53]: Can we double check the output of this HTML? (a nested img role inside of an img role, both with an aria-label) Maybe how we're passing ...rest props down or something?

Screenshot 2024-01-05 at 11 09 50 AM

It reads fine in VoiceOver, but it's the sort of weird HTML that might cause issues for screen readers.

**Review by hbuchel — COMMENTED** [review]
Thanks @esauerbo left a couple comments

File: packages/react/src/primitives/Alert/AlertIcon.tsx (hunk 2 above)
Comment (esauerbo) [reply packages/react/src/primitives/Alert/AlertIcon.tsx]: Should be fixed updated screenshot

File: packages/react-liveness/src/components/FaceLivenessDetector/shared/FaceLivenessErrorModal.tsx [hunk 3]
```diff
@@ -41,34 +42,40 @@ const renderToastErrorModal = (props: {
 
   let heading: string;
   let message: string;
+  let label: string;
 
   switch (errorState) {
     case LivenessErrorState.TIMEOUT:
       heading = timeoutHeaderText;
       message = timeoutMessageText;
+      label = errorLabelText;
```
Comment (thaddmt) [inline packages/react-liveness/src/components/FaceLivenessDetector/shared/FaceLivenessErrorModal.tsx:51]: small nit: if it's the same in all cases we can set the value outside of the switch statement right?

**Review by thaddmt — DISMISSED** [review]
LGTM

File: packages/react-liveness/src/components/FaceLivenessDetector/shared/FaceLivenessErrorModal.tsx (hunk 3 above)
Comment (reesscot) [reply packages/react-liveness/src/components/FaceLivenessDetector/shared/FaceLivenessErrorModal.tsx:51]: +1

**Review by thaddmt — DISMISSED** [review]

File: packages/react/src/primitives/Alert/AlertIcon.tsx [hunk 4]
```diff
@@ -14,6 +14,7 @@ import {
 interface AlertIconProps {
   variation?: AlertVariations;
   ariaHidden?: boolean;
+  role?: 'img';
```
[the remark this answers is not in the document]
Comment (calebpollman) [reply packages/react/src/primitives/Alert/AlertIcon.tsx:17]: We should extend the `role` property from `ViewProps` to align the property to the underlying icon component `role` definition, example:
```
interface AlertIconProps extends Pick<ViewProps, 'role'> {
  variation?: AlertVariations;
  ariaHidden?: boolean;
}
```

**Review by calebpollman — COMMENTED** [review]
Left some easy to address feedback. Thanks for doing this!

File: packages/react/src/primitives/Alert/AlertIcon.tsx [hunk 5]
```diff
@@ -11,9 +11,10 @@ import {
   useIcons,
 } from '../Icon';
 
-interface AlertIconProps {
+interface AlertIconProps extends Pick<ViewProps, 'role'> {
   variation?: AlertVariations;
   ariaHidden?: boolean;
+  ariaLabel?: string;
```
Comment (calebpollman) [inline packages/react/src/primitives/Alert/AlertIcon.tsx:17]: Pretty sure `ariaLabel` can be extended from `ViewProps`

**Review by calebpollman — APPROVED** [review]
🚢

**Review by thaddmt — APPROVED** [review]

**Pull request merged**